plugins/in_syslog/syslog_prot.c

```diff
@@ -31,88 +33,126 @@ static inline void consume_bytes(char *buf, int bytes, int length)
-    msgpack_unpacked   unpacked_buffer;
+    char              *modified_data_buffer;
+    int                modified_data_size;
+    msgpack_object_kv *new_map_entries[1];
+    msgpack_object_kv  message_entry;
     *result_buffer = NULL;
     *result_size = 0;
+    modified_data_buffer = NULL;
 
-    unpacker_offset = 0;
-    msgpack_unpacked_init(&unpacked_buffer);
-    result = msgpack_unpack_next(&unpacked_buffer,
-                                 base_object_buffer,
-                                 base_object_size,
-                                 &unpacker_offset);
-
-    if (result != MSGPACK_UNPACK_SUCCESS) {
-        return -1;
-    }
-
-    if (unpacker_offset != base_object_size) {
-        msgpack_unpacked_destroy(&unpacked_buffer);
-        return -2;
-    }
+    if (message_key_name != NULL) {
+        new_map_entries[0] = &message_entry;
 
-    if (unpacked_buffer.data.type != MSGPACK_OBJECT_MAP) {
-        msgpack_unpacked_destroy(&unpacked_buffer);
-        return -3;
-    }
+        message_entry.key.type = MSGPACK_OBJECT_STR;
+        message_entry.key.via.str.size = flb_sds_len(message_key_name);
+        message_entry.key.via.str.ptr  = message_key_name;
 
-    msgpack_sbuffer_init(&mp_sbuf);
-    msgpack_packer_init(&mp_pck, &mp_sbuf, msgpack_sbuffer_write);
+        if (message_type == MSGPACK_OBJECT_BIN) {
+            message_entry.val.type = MSGPACK_OBJECT_BIN;
+            message_entry.val.via.bin.size = message_size;
+            message_entry.val.via.bin.ptr  = message_buffer;
+        }
+        else if (message_type == MSGPACK_OBJECT_STR) {
+            message_entry.val.type = MSGPACK_OBJECT_STR;
+            message_entry.val.via.str.size = message_size;
+            message_entry.val.via.str.ptr  = message_buffer;
+        }
+        else {
+            result = FLB_MAP_NOT_MODIFIED;
 
-    msgpack_pack_map(&mp_pck, unpacked_buffer.data.via.map.size + 1);
+            return result;
+        }
 
-    for (i = 0; i < unpacked_buffer.data.via.map.size; i++) {
-        msgpack_pack_object(&mp_pck, unpacked_buffer.data.via.map.ptr[i].key);
-        msgpack_pack_object(&mp_pck, unpacked_buffer.data.via.map.ptr[i].val);
+        result = flb_msgpack_expand_map(base_object_buffer,
+                                        base_object_size,
+                                        new_map_entries, 1,
+                                        &modified_data_buffer,
+                                        &modified_data_size);
+        if (result == 0) {
+            result = FLB_MAP_EXPAND_SUCCESS;
+        }
     }
 
-    msgpack_pack_str(&mp_pck, flb_sds_len(raw_message_key_name));
-    msgpack_pack_str_body(&mp_pck, raw_message_key_name, flb_sds_len(raw_message_key_name));
-    msgpack_pack_str(&mp_pck, raw_message_size);
-    msgpack_pack_str_body(&mp_pck, raw_message_buffer, raw_message_size);
-
-    *result_buffer = mp_sbuf.data;
-    *result_size = mp_sbuf.size;
+    if (result != FLB_MAP_EXPAND_SUCCESS) {
```

Comment:
As per the comment in line 42 you need to invert the clause in this conditional block and remove the `result` override.

Then you need to add an else block to the clause in line 82 so for any other values returned by `flb_msgpack_expand_map` `result` is set to `FLB_MAP_EXPANSION_ERROR`.